File: Python_Samples/connect_customer_to_a_flow.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-

from pprint import pprint
import requests

from settings import sid, token


def connect_customer(sid, token,
                     customer_no, exotel_no, callerid, url,
                     timelimit=None, timeout=None, calltype="trans",
                     callback_url=None):
    return requests.post('https://twilix.exotel.in/v1/Accounts/{sid}/Calls/connect.json'.format(sid=sid),
        auth=(sid, token),
        data={
            'From': customer_no,
            'To': exotel_no,
            'CallerId': callerid,
            'Url': url,
            'TimeLimit': timelimit,
            'TimeOut': timeout,
            'CallType': calltype,
            'StatusCallback': callback_url
        })


if __name__ == '__main__':
    r = connect_customer(
        sid, token,
        customer_no="<Your-Customer's-Number>",
        exotel_no="<Your-Exotel-Landline-or-Mobile>",
        callerid="<Your-Exotel-virtual-number>",
        url="http://my.exotel.in/exoml/start/<flow_id>",
        timelimit="<time-in-seconds>",  # This is optional
        timeout="<time-in-seconds>",  # This is also optional
        calltype="trans",  # Can be "trans" for transactional and "promo" for promotional content
        callback_url="<http//: your company URL>"  # This is also also optional
        )
    print r.status_code
    pprint(r.json())

File: PHP_Samples/connect_customer_to_a_flow.php
<?php

$post_data = array(
    'From' => "<Your-Customer's-Number>",
    'To' => "<Your-Exotel-Landline-or-Mobile>",
    'CallerId' => "<Your-Exotel-virtual-number>",
    'Url' => "http://my.exotel.in/exoml/start/<flow_id>",
    'TimeLimit' => "<time-in-seconds>", //This is optional
    'TimeOut' => "<time-in-seconds>", //This is also optional
    'CallType' => "trans"
    'StatusCallback' => "<http//: your company URL>" //This is also also optional
);
 
// You can get your $exotel_sid, $api_key and $api_token from: https://my.exotel.com/apisettings/site#api-credentials 
$api_key = "xxxxx"; // Your `API KEY`.
$api_token = "xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx"; // Your `API TOKEN`
$exotel_sid = "xxxxxxxxxxx" // Your `Account Sid`
 
$url = "https://".$api_key.":".$api_token."@twilix.exotel.in/v1/Accounts/".$exotel_sid."/Calls/connect";
 
$ch = curl_init();
curl_setopt($ch, CURLOPT_VERBOSE, 1);
curl_setopt($ch, CURLOPT_URL, $url);
curl_setopt($ch, CURLOPT_POST, 1);
curl_setopt($ch, CURLOPT_RETURNTRANSFER, 1);
curl_setopt($ch, CURLOPT_FAILONERROR, 0);
curl_setopt($ch, CURLOPT_SSL_VERIFYPEER, 0);
curl_setopt($ch, CURLOPT_POSTFIELDS, http_build_query($post_data));
 
$http_result = curl_exec($ch);
$error = curl_error($ch);
$http_code = curl_getinfo($ch ,CURLINFO_HTTP_CODE);
 
curl_close($ch);
 
print "Response = ".print_r($http_result);

?>
